lib/src/arithmetic/floating_point/floating_point_adder_simple.dart

```diff
@@ -17,27 +17,31 @@ class FloatingPointAdderSimple<FpType extends FloatingPoint>
   /// Add two floating point numbers [a] and [b], returning result in [sum].
   /// - [adderGen] is an adder generator to be used in the primary adder
   /// functions.
-  /// - [priorityGen] is a [PriorityEncoder] generator to be used in the
-  /// leading one detection (default [RecursiveModulePriorityEncoder]).
   FloatingPointAdderSimple(super.a, super.b,
       {super.clk,
       super.reset,
       super.enable,
       Adder Function(Logic a, Logic b, {Logic? carryIn}) adderGen =
           NativeAdder.new,
-      PriorityEncoder Function(Logic bitVector, {Logic? valid, String name})
-          priorityGen = RecursiveModulePriorityEncoder.new,
+      List<int> Function(int) widthGen =
```

Comment:
doc comment on this `widthGen`

Reply:
thanks for adding the doc, but what's the purpose of it still?  Why would you want to use this?  What should a function return to be useful for this?  (this applies in all doc comments talking about this argument down the stack)